http/http_client.hpp

```diff
@@ -468,7 +467,7 @@ class ConnectionInfo : public std::enable_shared_from_this<ConnectionInfo>
             }
             if (connPolicy->retryPolicyAction == "RetryForever")
             {
-                state = ConnState::idle;
+                state = ConnState::closed;
```

Comment:
Why closed over terminate or suspend?

Reply:
@abhilashraju have we tested this change in retry forever scenario?
looks like this case needs to be tested.

Reply:
@raviteja-b yes I have done retry test . Doesnot show any issues in dev testing.
@gtmills  the  callback(false, connId, res); at 476  will set the connection state to closed anyway. do we really need this line of code. I don't know the history. Can we remove this code to simplify ?